File: FitFocus/FitFocus/FitFocus/Models/Workout.cs
using System;
using System.Collections.Generic;

namespace FitFocus.Models
{
    public class Workout
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string ApproxTime { get; set; }
        public List<Serie> Series;

        public Workout(string name, string approxTime, List<Serie> series = null)
        {
            if (series is null) series = new List<Serie>();

            this.Id = new Guid().ToString();
            this.Name = name;
            this.ApproxTime = approxTime;
            this.Series = series;
        }
    }

    public class Serie
    {
        public string Name;
        public int NumberOfSets;
        public DateTime TimeBetweenSets;
        public DateTime TimeAfterSets;
        public Excercise Excercise;

        public Serie(string name, int numberOfSets, DateTime TimeBetweenSets, DateTime TimeAfterSets, Excercise excercise)
        {
            this.Name = name;
            this.NumberOfSets = numberOfSets;
            this.TimeBetweenSets = TimeBetweenSets;
            this.TimeAfterSets = TimeAfterSets;
            this.Excercise = excercise;
        }
    }

    public class Excercise
    {
        public string Name;
        public string NumberOfRepetitions;

        public Excercise(string name, string numberOfRepetitions)
        {
            this.Name = name;
            this.NumberOfRepetitions = numberOfRepetitions;
        }
    }
}

File: PHP/Models/Workout.php
<?php



/**
 *
 */
class Workout
{
    public string $Name;
    public string $ApproxTime;
    public string $CoverImageUrl;
    public array $Series;

    /**
     * @throws WorkoutException
     */
    function FromJson($json):void{
        try {
            $this->Name = $json["Name"];
            $this->ApproxTime = $json["ApproxTime"];
            $this->CoverImageUrl = $json["CoverImageUrl"];
            $this->Series = [];
            foreach ($json["Series"] as $serieTmp) {
                include_once "Models/Serie.php";
                $serie = new Serie();
                $serie->Name = $serieTmp["Name"];
                $serie->NumberOfSets = $serieTmp["NumberOfSets"];
                $serie->TimeBetweenSets = $serieTmp["TimeBetweenSets"];
                $serie->TimeAfterSets = $serieTmp["TimeAfterSets"];
                include_once "Models/Exercise.php";
                $exercise = new Exercise();
                $exercise->Name = $serieTmp["Exercise"]["Name"];
                $exercise->NumberOfRepetitions = $serieTmp["Exercise"]["NumberOfRepetitions"];
                $exercise->PathToVideo = $serieTmp["Exercise"]["PathToVideo"];
                $exercise->PathToCoverImage = $serieTmp["Exercise"]["PathToCoverImage"];
                $serie->Exercise = $exercise;
                array_push($this->Series, $serie);
            }
        }catch(TypeError $e){
            throw new WorkoutException("Cannot create object from json");
        }
    }
}


class WorkoutException extends Exception{}